hash/structhash/structhash.go

```diff
@@ -91,15 +91,22 @@ func write(buf *bytes.Buffer, v reflect.Value) *bytes.Buffer {
 			buf.WriteString("nil")
 			return buf
 		}
-		if v.CanInterface() {
-			write(buf, v.Elem())
-		}
+		write(buf, v.Elem())
 	case reflect.Struct:
 		vt := v.Type()
 		items := make([]string, 0)
 		for i := 0; i < v.NumField(); i++ {
 			sf := vt.Field(i)
 			to := parseTag(sf)
+
+			// NOTE: structhash will allow to process all interface types.
+			// gogo/protobuf is not able to set tags for directly oneof interface.
+			// see: https://github.com/gogo/protobuf/issues/623
+			if to.name == "" && reflect.Zero(sf.Type).Kind() == reflect.Interface {
```

Comment:
Do I understand this correctly? This means that every single attribute that doesn't declare `hash:"name:X"` will be written in the hash if it's an interface. 

In that case, we really need to make sure that we never forgot any tag `hash` and also that all the generated attributes from proto `XXX_` are not impacting this and added in the hash when they shouldn't or we should add the ignore tag for these attributes.